dockstore-webservice/src/test/java/io/dockstore/webservice/TokenResourceTest.java

```diff
@@ -0,0 +1,198 @@
+        fakeUserinfoplus.setGender("New classification");
+        fakeUserinfoplus.setPicture("https://i.kym-cdn.com/photos/images/newsfeed/001/207/210/b22.jpg");
+        return fakeUserinfoplus;
+    }
+
+    private static Token getFakeToken() {
+        Token fakeToken = new Token();
+        fakeToken.setContent("fakeContent");
+        fakeToken.setTokenSource(TokenType.DOCKSTORE);
+        fakeToken.setUserId(1);
+        fakeToken.setUsername("fakeTokenUsername");
+        return fakeToken;
+    }
+
+    private static User getFakeUser() {
+        User fakeUser = new User();
+        fakeUser.setUsername("fakeUser");
+        fakeUser.setIsAdmin(false);
+        fakeUser.setAvatarUrl("https://i.kym-cdn.com/photos/images/newsfeed/001/207/210/b22.jpg");
+        fakeUser.setCurator(false);
+        return fakeUser;
+    }
+
+    @Before
+    public void setup() {
+        doReturn(getFakeTokenResponse()).when(tokenResourceSpy).getTokenResponse(anyString(), anyString());
+        doReturn(getFakeUserinfoplus()).when(tokenResourceSpy).getUserInfo(anyString());
+        doReturn(1l).when(tokenDAO).create(any());
+        doReturn(getFakeToken()).when(tokenDAO).findById(any());
+        doReturn(getFakeUser()).when(enduserDAO).findById(anyLong());
+    }
+
+    @After
+    public void tearDown() {
+        reset(tokenDAO);
+        reset(enduserDAO);
+        reset(tokenResourceSpy);
+    }
+
+    /**
+     * For a non-existing user, checks that two tokens (Dockstore and Google) were created
+     */
+    @Test
+    public void getGoogleTokenNewUser() {
+        // New user so return null
+        doReturn(null).when(enduserDAO).findByUsername(anyString());
+        Response potato = resources.target("/auth/tokens/google").request()
+                .header(javax.ws.rs.core.HttpHeaders.AUTHORIZATION, "Basic potato")
+                .post(Entity.json("{\n" + "  \"code\": \"fakeCode\",\n" + "  \"redirectUri\": \"fakeRedirectUri\"\n" + "}\n"));
+
+        // Two tokens should have been created, the Dockstore token and the Google Token
+        verify(tokenDAO, times(2)).create(any());
+        // Check profile is update once only once
+        verify(tokenResourceSpy, times(1)).updateGoogleUserMetaData(any(), any());
+        final String responseBody = potato.readEntity(String.class);
+        try {
+            Token token = mapper.readValue(responseBody, Token.class);
+            // Check that the token has the right info
+            Assert.assertEquals(1l, token.getUserId());
+            Assert.assertEquals("fakeContent", token.getContent());
+            Assert.assertEquals("fakeTokenUsername", token.getUsername());
+            Assert.assertEquals(TokenType.DOCKSTORE, token.getTokenSource());
+        } catch (IOException e) {
+            Assert.fail("Mapper problem: " + e.getMessage());
+        }
+    }
+
+    /**
+     * For an existing user without a Google token, checks that a token (Google) was created exactly once.
```

Comment:
Seems to check that a dockstore token was created?

Reply:
Checks that some token was created.  Checks that the Dockstore token is returned (GitHub endpoint also returns the Dockstore token)

Reply:
Ah ok, let's match the comment and check that the token created is actually a google token